### ISSUE TITLE
CPU power limits not sticking in ROG Zephyrus G16 (2024) GA605 AMD Ryzen 9 AI HX 370

### ISSUE THREAD

**Issue by 1ikhan** [seerge/g-helper#3017]
### Rules

- [X] I made myself familiar with the <a href='https://github.com/seerge/g-helper?tab=readme-ov-file'>Readme</a>, <a href='https://github.com/seerge/g-helper/wiki/FAQ'>FAQ</a> and <a href='https://github.com/seerge/g-helper/wiki/Troubleshooting'>Troubleshooting</a>.
- [X] I understand that, if insufficient information or no app logs will be provided, my issue will be closed without an answer.

### What's wrong?

Thanks for the free utility :)

When running Cinebench 2024, I move the power limit sliders around under Fans + Power with apply being checked, but the changes don't get reflected in HWInfo reporting, nor do the fans seem to slow if I set the power limit low to like 28W or something. For example, I start a 10-minute CB run in Balanced, start the run at 80W, and lower it from there but nothing changes in the power or temp reporting in HWInfo. HWInfo just reports like 60W throughout the entire slider range.

Interestingly, custom cpu fan curves and GPU clock limits work just fine.

### How to reproduce the bug?

1. Start CB24
2. Change power limit in ghelper
3. Wattage changes don't get reflected in HWInfo
4. Sadness

### Logs

[log.txt](https://github.com/user-attachments/files/16738540/log.txt)

### Device and Model

ROG Zephyrus G16 (2024) GA605 AMD Ryzen 9 AI HX 370

### Additional information.

Latest drivers and bios, etc. Armory Crate is uninstalled.

### Armoury Crate

Uninstalled

### Asus Services

7

### Version

0.184.0

### OS

Win 11 24H2

**Comment by seerge** [comment]
@1ikhan hello, what exactly are you looking at HWinfo ? It should be total CPU power draw being affected.

I see that app sets new power limits every time you drag a slider
```
8/24/2024 8:09:40 PM: PowerLimit A3 = 28 : OK
8/24/2024 8:09:40 PM: PowerLimit A0 = 28 : OK
8/24/2024 8:09:40 PM: PowerLimit C1 = 28 : OK
...
8/24/2024 8:11:47 PM: PowerLimit A3 = 80 : OK
8/24/2024 8:11:47 PM: PowerLimit A0 = 80 : OK
8/24/2024 8:11:47 PM: PowerLimit C1 = 80 : OK

...
8/24/2024 8:18:33 PM: PowerLimit A3 = 16 : OK
8/24/2024 8:18:33 PM: PowerLimit A0 = 80 : OK
8/24/2024 8:18:33 PM: PowerLimit C1 = 80 : OK
```

But keep in mind that when you have SPL set to 16W but  sPPT (2 min boost) set to 80W most likely it will keep using 80W for 2 minutes cause this is how any AMD CPU works. You need to adjust all 3 sliders :)

**Comment by 1ikhan** [comment]
Thanks for the reply! See attached screenshots
![image](https://github.com/user-attachments/assets/b9a5078e-ecf6-4618-a055-13643a43f8a4)
![image](https://github.com/user-attachments/assets/cad1809f-d0da-43b9-a11a-3ffb4a58bd86)
Those were done while not plugged in, but as you can see (unless I am looking at the wrong thing) the CPU package power figure does not change based on if it's on Balanced or Turbo, or 28W or 54W. When plugged in on Turbo (I don't have my adapter with me atm, I can send it in later if you'd like), the average goes to 80W. The power limits work as intended in armory crate, where in their custom performance profile I can set all the PLs to 54W and it actually shows up as 54W in HWinfo.

**Comment by seerge** [comment]
@1ikhan some models also require custom fan curves to be applied next to limits to make everything work (this is cause BIOS is designed for Armoury where you can't apply only power limits or only fan curves individually in Manual mode)

Can you try to click Apply under fan curves as well and see if it makes a difference ?

**Comment by 1ikhan** [comment]
Just did (the fan toggle is hidden but it's enabled)

![image](https://github.com/user-attachments/assets/3fb75880-e1ad-4ea4-b136-470f0c02196c)
![image](https://github.com/user-attachments/assets/48f55f79-95f2-43ec-a0b7-38f415260208)

Interestingly now, the lower power limit in Balanced sticks, but not the higher power limit in Turbo, and I suspect that will change if I plug in because there were different restrictions placed in the custom fan and power in AC for unplugged vs plugged.
In my unplugged tests here, Balanced stays around 28W like I set but Performance gets capped at 35W. I wonder what will happen if I plug in. Will update once I can.

**Comment by seerge** [comment]
@1ikhan correct,

Each mode (they are baked in BIOS), has own "hard" limits that you can't overcome. And this limits also differ depending if device is plugged / unplugged. So if it's programmed on firmware / bios level to stay on 35w for example on batter you can't force it to go higher.

But do I understand correctly that it was the fans checkbox that made a difference for power limits to start working ?

**Comment by 1ikhan** [comment]
@seerge Yep, the custom power limits were only respected after applying a custom fan curve and checking the "apply fan curve" box. Otherwise, the system appears to go to bios defaults when changing modes (~60w in balanced, 80w in turbo, fan behavior also changed depending on the profile when not applying any custom fan profiles).

**Comment by 1ikhan** [comment]
@seerge while I'm at it, I did some more testing, now for GPU, plugged in, balanced mode.

It appears that the power limit slider is more of a suggestion compared to clock limiting. Power limiting has more fluctuations in the power consumption, and increasing the limit does seem to have effects on scores, temps, and power consumption. Clock limit on the other hand, has more of a consistent effect on power consumption.

For example, let's say I want a GPU PL of 55W. The power limit slider set at 55W with no clock limit actually draws around 60W when CB24 is running, but clock limit set at 1800Mhz with power limit at max draws around 50-54W.

Are there any odd Nvidia quirks when limiting clocks?

**Comment by seerge** [comment]
@1ikhan GPU power limits is a new thing (introduced only in 2024 models). And this limit consists of 2 components base TDP and dynamic boost. So "real" power limit is base + boost together. 

Clock limiter - is a G-Helper specific feature (AC doesn't have that), it just uses nvidia driver (nvidia-smi app) to set a hard limit for clocks and works on all nvidia cards.

**Comment by 1ikhan** [comment]
@seerge Ah that makes sense, thanks. So, would it be fair to say that, if the goal is to optimize power consumption, then keeping all the power limits at max but reducing the clock limit slider is the way to go since it gives consistency?

**Comment by seerge** [comment]
@1ikhan clock limiter is just an alternative to power limits for device that don't support it :) or unless you pursue some specific goal.

This is pretty much same as disabling CPU Boost (not letting CPU go above base clocks) also limits power, but it's better just to set CPU power limits instead and let your CPU decide where and how to spend that power instead

**Comment by 1ikhan** [comment]
@seerge Thanks, and kudos for the awesome utility!

**Comment by seerge** [comment]
@1ikhan try this build, here custom power limits should work without checking apply under fans as well.

App would just automatically set  fan curve every time you select custom power limits. This is the "fix" G-Helper already does for other models also suffering from same behavior (including popular G14 2024 / 2023 for example)

Also I have restricted power slider range to 90w max (again same as it happens for all other popular models with AMD CPUs)

[GHelper.zip](https://github.com/user-attachments/files/16745768/GHelper.zip)

**Comment by 1ikhan** [comment]
@seerge appreciate the hard work! Although, I hope you don't mind me forgoing updating until the next stable release :)

Slightly unrelated, but if it helps for you to know more about the "PPW" of these new chips, I did some "vibe testing" in cyberpunk with optimized settings, Cinebench 24, and Geekbench 6.3, and it really does feel like leaving it to ASUS bios defaults is not the best move.

Observations:
- When testing the CPU only, disabling the dgpu: setting the max CPU power limit to 54W according to AMD's specs for this R9 370 APU. I think ASUS bios defaults to 80W max when not power limited, and programs like Cinebench 24 will gladly eat up nearly 80W at times, which reaches well into the diminishing returns for a highly negligible gain.
- When testing the GPU only, lowering the CPU power to like 28W: setting the max clock speed of the 4070 to Nvidia's specs of 2175MHz. Otherwise, ASUS bios will allow this to go up to 2505mhz, which again results in a highly negligible gain for more power and heat.
- When using both the CPU and GPU together: setting the max CPU power limit and max GPU clock limit as mentioned above results in a 3-4 fps gain in the built in benchmark of cyberpunk compared to an untouched turbo mode.

It's almost as if the designers of these components knew what they were doing when they finalized the specs... :suspect:

**Comment by cxw620** [comment]
> @1ikhan some models also require custom fan curves to be applied next to limits to make everything work (this is cause BIOS is designed for Armoury where you can't apply only power limits or only fan curves individually in Manual mode)
> 
> Can you try to click Apply under fan curves as well and see if it makes a difference ?

I met with the same problem and finally got the solution here. I suggest a kindly tip to remind users of such circumstance if possible.
